src/Listeners/AddApiAttributes.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of fof/nightmode.
+ *
+ * Copyright (c) 2020 FriendsOfFlarum.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FoF\Nightmode\Listeners;
+
+use Flarum\Api\Event\Serializing;
+use Flarum\Api\Serializer\ForumSerializer;
+use Flarum\Extension\ExtensionManager;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class AddApiAttributes
+{
+    /**
+     * @var SettingsRepositoryInterface
+     */
+    protected $settings;
+
+    /**
+     * @var ExtensionManager
+     */
+    protected $extensions;
+
+    public function __construct(SettingsRepositoryInterface $settings, ExtensionManager $extensions)
+    {
+        $this->settings = $settings;
+        $this->extensions = $extensions;
+    }
+
+    public function subscribe(Dispatcher $events)
+    {
+        $events->listen(Serializing::class, [$this, 'prepareApiAttributes']);
+    }
+
+    public function prepareApiAttributes(Serializing $event)
+    {
+        if ($event->isSerializer(ForumSerializer::class)) {
+            $event->attributes['fof-nightmode_default_theme'] = $this->getDefaultNightmodeTheme();
+        }
+    }
+
+    private function getDefaultNightmodeTheme(): int
+    {
+        return (int) app('flarum.settings')->get('fof-byobu.index_link', 0);
```

Comment:
I assume this is a copy paste error? Also, we can use the AddExtensionSettings extender from fof/extend instead: see
https://github.com/FriendsOfFlarum/prevent-necrobumping/blob/cfe4580e570bf38b5cbaf9584e61dfd96f2212b9/extend.php#L28-L30.

Reply:
I fixed that typo in the meantime. With it like that, I'm wondering why the changes worked in the first place! I have an idea though.

Thanks for pointing out that handy extender! I'll pop that in instead.

Reply:
@datitisev It looks like one of the downsides to this is that there's no way to specify an "ultra-default". For example, if someone closes the settings modal after enabling the extension, the setting added to the payload is `null`, where it's likely better to use `0`.

For now I'll add in null checks in the JS, but it would be a good idea to add support for this.

Reply:
Do you mean with the setting extender ? Yeah defaults & casting would be a good idea.